### PR TITLE
Refactoring of graphs overview tests

### PR DESCRIPTION
**Pull request by svilenvelikov**

### CODE REVIEW

File: test-cypress/integration/explore/graphs.overview.spec.js [hunk 1]
```diff
@@ -1,114 +1,125 @@
...
+
         cy.visit('/repository');
-        // Create new one
-        cy.createNewRepo(repoId).wait(2000);
-        cy.setRepoDefault(repoId);
-        cy.selectRepo(repoId);
+        cy.setRepoDefault(repositoryId);
+        cy.selectRepo(repositoryId);
 
-        // Import Wine dataset which will be used in most of the following tests
```
Comment (svilenvelikov) [inline test-cypress/integration/explore/graphs.overview.spec.js]: @mihailradkov  I'll refactor this to use the rest in another PR once you merge your PR.

File: test-cypress/integration/explore/graphs.overview.spec.js [hunk 2]
```diff
@@ -1,114 +1,131 @@
...
 
-    it('Test graphs overview search', function () {
-        cy.wait(1000);
-        cy.get('.form-control').type(graphToSearch);
-        cy.verifyGraphExists(graphToSearch);
-        cy.get("tbody").should('not.contain', "urn:11");
+    it('Test items per page switch menu', () => {
+        // Change number of items per page from the menu and verify that proper rows are present in
+        // the table.
+        selectItemFromMenu(1);
```
Comment (mihailradkov) [inline test-cypress/integration/explore/graphs.overview.spec.js]: It's not clear what 1 and 2 are, a comment can shed some light when reading the test. OR add properly named constants

File: test-cypress/integration/explore/graphs.overview.spec.js [hunk 3]
```diff
@@ -1,114 +1,131 @@
...
-            .each(($el, index) => {
-                if ($el.attr('title') === graphToDelete) {
-                    cy.get('[class="fa fa-trash-o fa-lg"]').eq(index).click({force: true});
-                    cy.get('.modal-content button.btn.btn-primary').click({force: true});
-                }
-            });
-        cy.get("tbody").should('not.contain', graphToDelete);
+    it('Test graphs overview search', () => {
+        // Default items/rows per page is 10.
+        verifyVisibleGraphsCountIs(10);
```
Comment (mihailradkov) [inline test-cypress/integration/explore/graphs.overview.spec.js]: `verifyVisibleGraphsCount` ?

File: test-cypress/integration/explore/graphs.overview.spec.js [hunk 4]
```diff
@@ -1,114 +1,131 @@
...
-        cy.get(".modal-content button.btn.btn-primary").click();
-        cy.get("tbody").should('not.contain', "urn");
-        cy.verifyGraphExists("The default graph");
-
+    it('Delete graph', () => {
+        let graphToDelete = 'urn:9';
+        cy.get(`${EXPORT_GRAPHS_TABLE_ID} tbody a[title="${graphToDelete}"]`)
+            .closest('tr').find('.delete-graph-btn').click();
+        confirmDelete();
+        verifyGraph(graphToDelete, false);
```
Comment (mihailradkov) [inline test-cypress/integration/explore/graphs.overview.spec.js]: Well I forgot what `verifyGraph` does exactly and it's not clear what is verified here. Probably another method `verifyGraphIsMissing`  or rename to `verifyGraphExistence` ?

File: test-cypress/integration/explore/graphs.overview.spec.js [hunk 5]
```diff
@@ -1,114 +1,131 @@
...
-        cy.get("li").should('contain', "3").should('be', 'active')
-        cy.verifyGraphExists("urn:21");
-    });
+    //verifies that a certain graph exist within the current page in graphs overview.
+    function verifyGraph(graphName, exists = true) {
+        cy.get(EXPORT_GRAPHS_TABLE_ID).should(`${!exists ? 'not.' : ''}contain`, graphName);
+    }
 
+    function selectPage(number) {
+        return cy.get('div[paginations] ul li a').eq(number).click();
```
[the remark this answers is not in the document]
Comment (svilenvelikov) [reply test-cypress/integration/explore/graphs.overview.spec.js]: That's true. The thing is that in the pager along with the page links are present also the `first` and `last` links. So this selector works properly due to that fact. I'll see if I can fix it in order to not be as misleading as it is now.

File: test-cypress/integration/explore/graphs.overview.spec.js (hunk 3 above)
Comment (svilenvelikov) [reply test-cypress/integration/explore/graphs.overview.spec.js]: Somehow this seemed more natural to read to me :)

File: test-cypress/integration/explore/graphs.overview.spec.js (hunk 4 above)
Comment (svilenvelikov) [reply test-cypress/integration/explore/graphs.overview.spec.js]: You are right, this name is not very descriptive. I'll use the `verifyGraphExistence`

**Pull request merged**